src/Chapter3.hs

```diff
@@ -970,6 +1111,44 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data WeekDay
+	= Monday
+	| Tuesday
+	| Wednesday
+	| Thursday
+	| Friday
+	| Saturday
+	| Sunday deriving (Show, Read, Eq, Ord, Enum)
+
+isWeekend :: WeekDay -> Bool
+isWeekend day = case day of
+	Saturday -> True
+	Sunday -> True
+	_ -> False
+
+nextDay :: WeekDay -> WeekDay
+nextDay day = case day of
+	Sunday -> Monday
+	_ -> succ day
+
+daysToParty :: WeekDay -> Int
+daysToParty day
+	| (day > Friday) = 7 - (fromEnum day - fromEnum Friday)
+	| otherwise = fromEnum Friday - fromEnum day
+
```

Comment:
Nice solution! If you remember from Chapter One, how `mod` works for negative numbers, you can probably implement a shorter solution :slightly_smiling_face: